kernel/init.c

```diff
@@ -108,6 +108,15 @@ void __weak z_early_memcpy(void *dst, const void *src, size_t n)
 __boot_func
 void z_bss_zero(void)
 {
+	if (IS_ENABLED(CONFIG_ARCH_POSIX)) {
```

Comment:
I'm a bit surprised that you needed this change. From my recollection and from looking at the code , z_bss_zero was called from arch code (prep_c ) but for the posix arch we never did so. Looking now at the code I can see this is also called a 2nd time from z_mem_manage_init() if CONFIG_MMU is set, but that should not be the case for the posix arch either. Anyhow, this change cannot harm.

Reply:
It's not called.  But neither is it dropped from the build (presumably because __boot_func defeats -ffunction-sections and it won't be GC'd), so it's presented to the linker where it dies on the undefined symbol that clang doesn't emit.

Reply:
Oh :) I see. I'm ok with that return being there, or the function being ifndef'ed or.. whatever you think is best.
By the way thanks for keeping the xSAN alive and adding MSAN.